### PR TITLE
Property tests for SHOC Length Scale lower level functions

### PR DESCRIPTION
**Pull request by bogensch**
Add property tests and boiler plate code for the lower level functions called by shoc_length() function. Specifically, tests for the following functions have been added (see *cpp file for specific description of each test):

1.  compute_brunt_shoc_length
2.  compute_l_inf_shoc_length
3.  compute_conv_vel_shoc_length
4.  compute_conv_time_shoc_length
5.  compute_shoc_mix_shoc_length
6.  check_length_scale_shoc_length

### CODE REVIEW

**Comment by bogensch** [comment]
setting as WIP in light of PR #499.  Once that is on master I will update.  Hold off on reviews for now.

**Comment by bogensch** [comment]
This PR is now ready for review.

**Comment by jgfouca** [comment]
@bogensch , I reset your branch to the point before it merged in the stuff we didn't want (#501)

File: components/scream/src/physics/shoc/tests/shoc_brunt_length_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+#include "physics/share/physics_constants.hpp"
+#include "physics/shoc/shoc_functions.hpp"
+#include "physics/shoc/shoc_functions_f90.hpp"
+#include "shoc_unit_tests_common.hpp"
+
+namespace scream {
+namespace shoc {
+namespace unit_test {
+
+TEST_CASE("shoc_brunt_length", "shoc") {
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/tests/shoc_brunt_length_tests.cpp:24]: In order to prepare these for BFB testing, please use the pattern that I did for calc-vertflux. Each unit test file should have a struct that has static methods for property and bfb tests.

Comment (bogensch) [reply components/scream/src/physics/shoc/tests/shoc_brunt_length_tests.cpp]: Thanks @jgfouca I completely missed that!  All *.cpp files in relation to this PR have been updated for your review.  I realize all other .cpp testing files (besides your calc-vertflux) are still stuck in the old pattern and need to be updated.  Figure maybe I should do those and update them in this PR, unless you prefer I do that in a separate PR?

Comment (jgfouca) [reply components/scream/src/physics/shoc/tests/shoc_brunt_length_tests.cpp]: @bogensch , whatever is easiest for you.

**Pull request merged**